src/api/index.js

```diff
@@ -0,0 +1,46 @@
+const baseUrl = 'https://nc-news-t20n.onrender.com/api';
+
+export const getAllArticles = () => {
+  return apiGet('/articles');
+};
+
+const api = (url, options = {}) => {
+  return fetch(`${baseUrl}${url}`, { ...options })
+    .then((response) => {
+      return response.status !== 204 ? response.json() : { success: true };
+    })
+};
+
+const apiGet = (url, options = {}) => {
+  const { query = null } = options;
+
+  let queryStr = '';
+  if (query) {
+    queryStr = query.toString();
+    queryStr = queryStr && `?${queryStr}`;
+  }
+
+  return api(`${url}${queryStr}`);
+};
+
+const apiPost = (url, body) => {
+  return api(url, {
+    method: 'POST',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify(body),
+  });
+};
+
+const apiPatch = (url, body) => {
+  return api(url, {
+    method: 'PATCH',
+    headers: { 'Content-Type': 'application/json' },
+    body: JSON.stringify(body),
+  });
+};
+
+const apiDelete = (url) => {
+  return api(url, {
+    method: 'DELETE',
+  });
+};
```

Comment:
I like that you've got these function separated out into a separate file - I see you've opted to use `fetch` instead of axios which I don't hate - but a lot of functionality you've had to make manually here comes as standard in axios. I would recommend using it in this scenario, but it's up to you!
Also there's code here that isn't needed for this particular ticket (e.g. the post and delete functionality). It's good to get in the habit of just covering functionality specified by the ticket (this becomes even more important in projects with multiple people to avoid duplicating code)

Reply:
I spent a lot of time perfecting this mini lib from the previous sprint so copied in verbatim but I did consider leaving unused methods out until required.

Deleted them for now.